### PR TITLE
terraform: move latest to BUSL version, add :1.5.7 for last pre-BUSL version

### PR DESCRIPTION
**Pull request by imjasonh**
Ran locally

```
module.terraform.oci_tag.latest-busl: Creation complete after 1s [id=ttl.sh/jason/terraform:latest@sha256:2964ce66f290c759abda053b343d297df39555bfc6fcf8cc09de32d2e5c03acb]
module.terraform.oci_tag.latest-busl-dev: Creation complete after 1s [id=ttl.sh/jason/terraform:latest-dev@sha256:f5e69646e91c6db07a1976435faa0f66e8351749aabcd4266b6dde712a69bd75]
module.terraform.oci_tag.pre-busl-dev["1.5"]: Creation complete after 2s [id=ttl.sh/jason/terraform:1.5-dev@sha256:f5e69646e91c6db07a1976435faa0f66e8351749aabcd4266b6dde712a69bd75]
module.terraform.oci_tag.pre-busl["1.5.7"]: Creation complete after 2s [id=ttl.sh/jason/terraform:1.5.7@sha256:2964ce66f290c759abda053b343d297df39555bfc6fcf8cc09de32d2e5c03acb]
module.terraform.oci_tag.pre-busl-dev["1.5.7"]: Creation complete after 2s [id=ttl.sh/jason/terraform:1.5.7-dev@sha256:f5e69646e91c6db07a1976435faa0f66e8351749aabcd4266b6dde712a69bd75]
module.terraform.oci_tag.pre-busl["1.5"]: Creation complete after 3s [id=ttl.sh/jason/terraform:1.5@sha256:2964ce66f290c759abda053b343d297df39555bfc6fcf8cc09de32d2e5c03acb]
```

### CODE REVIEW

**Comment by dlorenc** [comment]
You're my hero

File: images/terraform/main.tf [hunk 1]
```diff
@@ -8,32 +8,66 @@ variable "target_repository" {
...
-resource "oci_tag" "latest-dev" {
-  depends_on = [module.test-latest]
-  digest_ref = module.latest.dev_ref
+resource "oci_tag" "latest-busl-dev" {
+  depends_on = [module.test-latest-busl]
+  digest_ref = module.latest-busl.dev_ref
   tag        = "latest-dev"
 }
+
+locals { pre_busl_tags = toset(["1.5.7", "1.5"]) }
```
Comment (joshrwolf) [inline images/terraform/main.tf:59]:
```suggestion
locals { pre_busl_tags = toset(["1.5.7", "1.5", "latest-busl", "latest-busl-dev"]) }
```

would these be useful? I imagine most folks don't know or want to know that 1.5.7 is the latest prebusl

Comment (joshrwolf) [reply images/terraform/main.tf:59]: I guess these need auth exceptions so don't hold up on this comment!

**Review by joshrwolf — APPROVED** [review]

File: images/terraform/main.tf (hunk 1 above)
Comment (imjasonh) [reply images/terraform/main.tf:59]: This might be useful, but these are the _pre_-busl images. 

Maybe `latest-non-busl(-dev)`?

wdyt @dlorenc

Comment (dlorenc) [reply images/terraform/main.tf:59]: maybe latest-mpl?

**Comment by imjasonh** [comment]
```
module.terraform.oci_tag.latest-busl-dev: Creation complete after 1s [id=ttl.sh/jason/terraform:latest-dev@sha256:14a65ade133ce1c9e745a119e20359cd247a5fec977e4d02dcb34bfecd05a1a7]
module.terraform.oci_tag.pre-busl["1.5.7"]: Creation complete after 1s [id=ttl.sh/jason/terraform:1.5.7@sha256:2964ce66f290c759abda053b343d297df39555bfc6fcf8cc09de32d2e5c03acb]
module.terraform.oci_tag.latest-busl: Creation complete after 1s [id=ttl.sh/jason/terraform:latest@sha256:2964ce66f290c759abda053b343d297df39555bfc6fcf8cc09de32d2e5c03acb]
module.terraform.oci_tag.pre-busl-dev["1.5.7"]: Creation complete after 1s [id=ttl.sh/jason/terraform:1.5.7-dev@sha256:14a65ade133ce1c9e745a119e20359cd247a5fec977e4d02dcb34bfecd05a1a7]
module.terraform.oci_tag.pre-busl["1.5"]: Creation complete after 1s [id=ttl.sh/jason/terraform:1.5@sha256:2964ce66f290c759abda053b343d297df39555bfc6fcf8cc09de32d2e5c03acb]
module.terraform.oci_tag.pre-busl-dev["1.5"]: Creation complete after 1s [id=ttl.sh/jason/terraform:1.5-dev@sha256:14a65ade133ce1c9e745a119e20359cd247a5fec977e4d02dcb34bfecd05a1a7]
module.terraform.oci_tag.pre-busl["latest-mpl"]: Creation complete after 2s [id=ttl.sh/jason/terraform:latest-mpl@sha256:2964ce66f290c759abda053b343d297df39555bfc6fcf8cc09de32d2e5c03acb]
module.terraform.oci_tag.pre-busl-dev["latest-mpl"]: Creation complete after 2s [id=ttl.sh/jason/terraform:latest-mpl-dev@sha256:14a65ade133ce1c9e745a119e20359cd247a5fec977e4d02dcb34bfecd05a1a7]
```

**Review by dlorenc — APPROVED** [review]

**Pull request merged**